org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/ISourceLookUpProvider.java

```diff
@@ -0,0 +1,13 @@
+package org.eclipse.jdt.ls.debug.adapter;
+
+import java.util.Map;
+
+public interface ISourceLookUpProvider {
+    void initializeContext(Map<String, Object> props);
```

Comment:
If all provider shares the same initialize routine, make it another interface and let others extend it.